packages/core/src/machines/authMachine.ts

```diff
@@ -0,0 +1,172 @@
+    iframe: false, // open in new window
+  });
+}
+
+export const authMachine = createMachine<AuthContext, AuthEvent>(
+  {
+    id: 'auth',
+    initial: 'idle',
+    context: {},
+    states: {
+      // See: https://xstate.js.org/docs/guides/communication.html#invoking-promises
+      idle: {
+        invoke: [
+          {
+            // TODO Wait for Auth to be configured
+            src: 'getCurrentUser',
+            onDone: {
+              actions: 'setUser',
+              target: 'authenticated',
+            },
+            onError: 'signIn',
+          },
+          {
+            src: 'getAmplifyConfig',
+            onDone: {
+              actions: 'setAuthConfig',
+            },
+          },
+        ],
+      },
+      signIn: {
+        entry: 'spawnSignInActor',
+        exit: stopActor('signInActor'),
+        on: {
+          SIGN_UP: 'signUp',
+          RESET_PASSWORD: 'resetPassword',
+          'done.invoke.signInActor': [
+            {
+              target: 'signUp',
+              cond: 'shouldRedirectToSignUp',
+            },
+            {
+              target: 'authenticated',
+              actions: 'setUser',
+            },
+          ],
+        },
+      },
+      signUp: {
+        entry: 'spawnSignUpActor',
+        exit: stopActor('signUpActor'),
+        on: {
+          SIGN_IN: 'signIn',
+          'done.invoke.signUpActor': {
+            target: 'signIn',
+            actions: 'setUser',
+          },
+        },
+      },
+      resetPassword: {
+        entry: 'spawnResetPasswordActor',
+        exit: stopActor('resetPasswordActor'),
+        on: {
+          SIGN_IN: 'signIn',
+          'done.invoke.resetPasswordActor': 'signIn',
+        },
+      },
+      signOut: {
+        entry: 'spawnSignOutActor',
+        exit: [stopActor('signOutActor'), 'clearUser'],
+        on: { 'done.invoke.signOutActor': 'idle' },
+      },
+      authenticated: {
+        on: { SIGN_OUT: 'signOut' },
+      },
+    },
+    on: {
+      CHANGE: { actions: 'forwardToActor' },
+      SUBMIT: { actions: 'forwardToActor' },
+      FEDERATED_SIGN_IN: { actions: 'forwardToActor' },
+      RESEND: { actions: 'forwardToActor' },
+      SIGN_OUT: { actions: 'forwardToActor' },
+    },
+  },
+  {
+    actions: {
+      forwardToActor: forwardTo((context) => context.actorRef),
+      setUser: assign({
+        user: (_, event) => event.data.user || event.data,
+      }),
+      clearUser: assign({
+        user: undefined,
+      }),
+      setAuthConfig: assign({
+        config(_, event) {
+          return event.data.auth;
+        },
+      }),
+      spawnSignInActor: assign({
+        actorRef: (_, event) => {
+          const actor = signInActor.withContext({
+            authAttributes: event.data?.authAttributes,
+            user: event.data?.user,
+            intent: event.data?.intent,
+            formValues: {},
+            validationError: {},
+          });
```

Comment:
This is how we pass required context to actors. Note that `event.data` is coming from actor `done` events.